lib/data/models/question.dart

```diff
@@ -0,0 +1,17 @@
+import 'package:job_interview_practice/domain/repository/question.dart';
+
+class QuestionModel extends QuestionRepository{
```

Comment:
This should extend `Question` (in `/lib/domain/entities/question.dart`) not `QuestionRepository`.  The purpose of having a `QuestionModel` is to introduce external-interface (web, db, etc) concerns, keeping the Entity pure of them.  See https://tpa-sendai.slack.com/archives/C01DZ8F1P1S/p1605562201075800

Now, regarding repositories -- the implementation of a repository should live under `data/repositories`.  

It would be called `QuestionRepositoryImpl`, and live in a file called `question_repository_impl.dart`.  

It should use the `implement` keyword to indicate that it is implementing an interface. 

```
class QuestionRepositoryImpl implements QuestionRepository {

```

The concrete repository implementation is what ties models to external data sources. 

So `QuestionRepositoryImpl` would import and use the `QuestionModel`.

Reply:
Thank  you for this review! This makes me clear about archtecture.
So I'll change repository-dir from /domain/ to /data/ , and let it use `QuestionModel`